### ISSUE TITLE
Docker-compose not accepting 'extends' option in docker-compose.yml file

### ISSUE THREAD

**Issue by sunnysingh1985** [docker/compose#1245]
Given a "docker-compose.yml":

```yml
web:
 extends:
  file: common.yml
  service: webapp
```
and another "common.yml":
```yml
webapp:
 image: ubuntu
```
Executing 'docker-compose ps' throws following error:

```sh
$ docker-compose ps
$ Unsupported config option for web service: 'extends'
```

**Comment by stephengfriend** [comment]
@sunnysingh1985 What version of docker-compose are you using? I just ran into this issue to realize the that extends isn't added until 1.2.0rc1, so unless you're using a pre-release version, it won't work.

**Comment by sunnysingh1985** [comment]
@stephengfriend - thanks for the update! I didn't know that it was not part of "docker-compose 1.1.0". I saw this listed on the docs and I was tempted to use it :) Thanks anyways, I will wait for this.

**Comment by asmaps** [comment]
Maybe there should be a note in the docs when the command was added and/or only show the docs for the current release on docs.docker.com

**Comment by KingScooty** [comment]
Just stumbled across this too. Did exactly what @sunnysingh1985 did and thought i'd try it out after seeing it in the docs. 

What's another way to split the config up into development, testing, production without installing docker-composer 1.2.0rc1?

**Comment by stephengfriend** [comment]
@KingScooty Even using the extends option from 1.2.0rc1, I've found that I basically have to keep multiple .yml files, with much duplication, for each environment. I've been using docker-compose.yml for "production" and docker-compose.<env>.yml, where <env> is dev or test. Then just pass the -f flag if in dev or test.

Using extends, all I've been able to gain is an extra docker-compose.common.yml which has some base entries in it and then mounting volumes, volumes_from, linking, environment stuff is configured in the appropriate environment .yml file.

**Comment by KingScooty** [comment]
Can't seem to install 1.2.0 on circleCI anyway, bah.
@stephengfriend Ah right, so it's not actually that special then?

So how do you switch between those 2 configs? I really need to setup a testing config file to tell docker to set net to host in order to use the preinstalled Redis on circleCI.

Ideally i'd be switching between docker-compose.yml and testing.docker-compose.yml.

**Comment by dnephin** [comment]
This should be resolved since we did this release a while ago.

The multiple compose files should be addressed by #2051 which will be in the 1.5.0 release
